op-devstack/dsl/l2_el.go

```
@@ -357,6 +357,34 @@ func (el *L2ELNode) MatchedUnsafe(refNode SyncStatusProvider, attempts int) {
 	el.Matched(refNode, types.LocalUnsafe, attempts)
 }
 
+// PendingNonceMatchedFn returns a lambda that checks that the pending nonce of an account matches the provided reference nonce
+func (el *L2ELNode) PendingNonceMatchedFn(account common.Address, nonce uint64, attempts int, duration time.Duration) CheckFunc {
+	return func() error {
+		logger := el.log.With("id", el.inner.ID(), "account", account)
+		logger.Debug("Expecting pending nonce to match with reference nonce", "nonce", nonce)
+		return retry.Do0(el.ctx, attempts, &retry.FixedStrategy{Dur: duration},
+			func() error {
+				baseNonce, err := el.inner.EthClient().PendingNonceAt(el.ctx, account)
+				if err != nil {
+					return fmt.Errorf("failed to get pending nonce from node: %w", err)
+				}
+
+				if baseNonce == nonce {
+					logger.Debug("Pending nonce matched", "nonce", baseNonce)
+					return nil
+				}
```

Is it worth checking to see if `baseNonce > nonce` and failing faster in that case?

---

We care about an exact nonce - there shouldn't be any other actor submitting transactions with the address that is being checked.